cfme/tests/intelligence/reports/test_canned_corresponds.py

```diff
@@ -15,7 +16,10 @@
 
 @pytest.fixture(scope="module")
 def setup_a_provider():
-    setup_a_provider_by_class(InfraProvider)
+    try:
```

Comment:
I think you want to refactor this a bit to generate tests differently using the provider filter.

By default testgen.generate uses testgen.providers_by_class as gen_func, which takes a list of classes. You can set the scope to module, and then use pytestmark to add the setup_provider fixture for all of the tests instead of specifying the fixture in each test's definition.

Try this out:
```python
 pytest_generate_tests = testgen.generate(classes=[InfraProvider], scope='module')

pytestmark = [ pytest.mark.usefixtures('setup_provider') ]
```